applications/flash_attention_v2/collective/xe_flash_attn_epilogue.hpp

```diff
@@ -201,14 +202,37 @@ class CollectiveEpilogueAttention<IntelPVCEpilogue, CtaTileMNK_, ElementO_, Stri
     copy(params.xe_store_o, out, tOgO);
   }
 
+  template <bool VarLen, class ProblemShapeType>
+  CUTLASS_DEVICE static constexpr Params get_updated_copies(Params const& params, ProblemShapeType const& problem_shape, int const& l_coord) {
+    if constexpr (!VarLen) {
+      return params;
+    } else {
+      auto [batch, num_heads, seq_len_qo, seq_len_kv, head_size_qk, head_size_vo] = problem_shape;
+
+      auto qo_cumulative_length = get<2>(problem_shape).cumulative_length;
+      int offset_o = l_coord == 0 ? 0 : num_heads * head_size_vo * qo_cumulative_length[l_coord];
+      XE_Copy_O xe_store_o = {};
+      auto store_traits = static_cast<Trait_O const&>(params.xe_store_o);
+      ElementO* base_ptr = (ElementO*)store_traits.base_ptr;
+      xe_store_o = make_tiled_copy(Copy_Atom<Trait_O, ElementO>{}.with(
+                                        make_tensor(make_gmem_ptr(base_ptr + offset_o),
+                                                    make_layout(make_shape(seq_len_qo, head_size_vo, num_heads),
+                                                                make_stride(static_cast<int64_t>(head_size_vo), cute::C<1>{}, static_cast<int64_t>(seq_len_qo * head_size_vo))))),
+                                  Layout<Shape<_1, Int<SubgroupSize>>>{},
+                                  make_layout(make_shape(get<0>(typename Trait_O::BlockShape{}),
+                                                          get<1>(typename Trait_O::BlockShape{}) / Int<SubgroupSize>{})));
```

Comment:
you can make this part of the code nicer if you make a variable with copy thread shape and use shape_div (see xe_mma)

Reply:
Or even better I think you can use the code like in my next suggestion.